cms/djangoapps/contentstore/management/commands/import_content_library.py

```diff
@@ -0,0 +1,133 @@
+"""
+Script for importing a content library from a tar.gz file
+"""
+from __future__ import print_function
+
+import base64
+import os
+import tarfile
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.core.exceptions import SuspiciousOperation
+from django.core.management.base import BaseCommand, CommandError
+from lxml import etree
+from opaque_keys.edx.keys import CourseKey
+from path import Path
+from xmodule.contentstore.django import contentstore
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.exceptions import DuplicateCourseError
+from xmodule.modulestore.xml_importer import import_library_from_xml
+
+from cms.djangoapps.contentstore.utils import add_instructor
+from openedx.core.lib.extract_tar import safetar_extractall
+
+
+class Command(BaseCommand):
+    """
+    Import the specified content library archive.
+    """
+    help = 'Import the specified content library into mongo'
+
+    def add_arguments(self, parser):
+        parser.add_argument('archive_path')
+        parser.add_argument('owner_username')
+
+    def handle(self, *args, **options):
+        """
+        Given a content library archive path, import the corresponding course to mongo.
+        """
+
+        archive_path = options['archive_path']
+        username = options['owner_username']
+
+        data_root = Path(settings.GITHUB_REPO_ROOT)
+        subdir = base64.urlsafe_b64encode(repr(archive_path.split('/')[-1]))
```

Comment:
Is this `split('/')` the same as `os.path.basename`?

Why is the basename `repr()`ed here? Is this just used as temporary directory? We shouldn't need to add the quotes that `repr()` would add to a string if so.

Reply:
Most of the import code was extracted from Studio and I refrained from modifying due to time constraints working on the original scripts. But I agree it's a good idea fixing it now.